src/modules/blockchainApplication/explore/components/BlockchainApplicationList/BlockchainApplicationList.js

```diff
@@ -0,0 +1,116 @@
+import React, {
+  useCallback, useMemo, useRef, useState,
+} from 'react';
+import { useTranslation } from 'react-i18next';
+import BoxHeader from 'src/theme/box/header';
+import Box from 'src/theme/box';
+import BoxContent from 'src/theme/box/content';
+import { usePinBlockchainApplication } from '@blockchainApplication/manage/hooks/usePinBlockchainApplication';
+import grid from 'flexboxgrid/dist/flexboxgrid.css';
+import Table from 'src/theme/table';
+import { Input } from 'src/theme';
+import Icon from 'src/theme/Icon';
+import BlockchainApplicationRow from '../BlockchainApplicationRow';
+import header from './BlockchainApplicationListHeaderMap';
+import styles from './BlockchainApplicationList.css';
+import { BLOCKCHAIN_APPLICATION_LIST_LIMIT } from '../../const/constants';
+import BlockchainApplicationSkeleton from '../BlockchainApplicationSkeleton';
+
+// eslint-disable-next-line max-statements
+const BlockchainApplicationList = ({
+  applications: apps,
+  applyFilters,
+  filters,
+}) => {
+  const [searchValue, setSearchValue] = useState('');
+  const debounceTimeout = useRef(null);
+  const { t } = useTranslation();
+  const { pins, checkPinByChainId, togglePin } = usePinBlockchainApplication();
+
+  const applications = useMemo(() => {
+    if (pins.length && apps.data.length) {
+      return {
+        ...apps,
+        data: apps.data.map(chainData => ({
+          ...chainData,
+          isPinned: checkPinByChainId(chainData.chainID),
+        }
+        )),
+      };
+    }
+    return apps;
+  }, [apps, checkPinByChainId]);
```

Comment:
missing pins dependencies
```suggestion
  }, [apps, pins, checkPinByChainId]);
```

Reply:
because `checkPinByChainId` depends on pins i.e 
<img width="636" alt="image" src="https://user-images.githubusercontent.com/32064769/178284603-de6688f4-2d15-4602-af77-fb5380ae42e9.png">

the useEffect will always be called when there is a change in `pins` because a new instance of the checkPinByChainId function would be created on every change in `pins`

Reply:
just FYI: with the right enlist config that one can generate an error